### PR TITLE
Header update, mobile responsive

### PR DESCRIPTION
**Pull request by AvocadoCasualty**
desktop-header-dropdown
header-desktop
mobile-header
mobile-header-menu

### CODE REVIEW

File: public/src/shared/_header.scss [hunk 1]
```diff
@@ -1,4 +1,204 @@
...
+          text-align: left;
+        }
+      }
+
+      .children {
+        list-style: none;
+        height: 100%;
+        text-align: left;
+
+        a {
```
Comment (thorbaert) [inline public/src/shared/_header.scss]: Personally I think the text is kinda dark I'd thinking we maybe make it the same color as the rest and see if peter is cool with it?

File: public/src/shared/header.js [hunk 2]
```diff
@@ -1,11 +1,193 @@
+import {useState} from "react"
 import {Link} from "react-router-dom";
+import {AiOutlineMenu, BsBag, IoPersonCircle, GrClose} from "react-icons/all";
 
 function Header() {
+    const [isShown, setIsShown] = useState({})
+    const [sideMenu, setSideMenu] = useState(false)
+    const [dropdownOpen, setDropdownOpen] = useState(false)
+
+    const links = [
```
Comment (thorbaert) [inline public/src/shared/header.js]: I would move this outside of the function. Not much time in declaration but it helps not to be declared every time the component changes.

**Review by thorbaert — CHANGES_REQUESTED** [review]

File: public/src/shared/header.js (hunk 2 above)
Comment (AvocadoCasualty) [reply public/src/shared/header.js]: Smart. Didn't even think about it as it'll be replaced when we get a backend to link to. I'll fix it!

File: public/src/shared/_header.scss (hunk 1 above)
Comment (AvocadoCasualty) [reply public/src/shared/_header.scss]: Sure! Easy peasy

**Review by thorbaert — APPROVED** [review]

**Pull request merged**